Improving General Readability and Usability

ifBars opened the pull request:
- The major changes are the comments added, some to specific lines, some to groups
- Some hard coded values like, the fan's minimum and maximum speeds, along with a few others, are now stored as float variables
- Some variable names were changed to improve readability, like h for example was changed to currentHumidity, as that is what it represents
- The SET_TEMP command has been split into 2 separate commands, SET_TEMPF and SET_TEMPC, for celsius and fahrenheit, to allow for more usability worldwide
- Conversion of celsius to fahrenheit is handled in a separate function to allow for modular usage in the SET_TEMPF and SET_TEMPC commands

Edit 9/15/23: 
- Added a feature to track the average temp and humidity in the past hour, which is displayed upon the DATAREQ command and printed to serial every 2 seconds.
- Introduced a small, adjustable, 3 second delay between state changes (Ex: turning on/off devices or adjusting fan speed). This prevents rapid toggling and allows devices to stabilize before making further adjustments.
- Changed the hysteresis values to constants to allow for reusability and easy adjustability
- Added more comments and improved some others

Edit 9/18/23
- Unsigned the samplesPerHour variable to prevent negative values
- Added debug messages that are togglable with the debug boolean on line 31
- Added a function to format DateTime to a string
- Added a function to calculate current dew point, and also added current dew point to DATAREQ command
- Updated code to make sure it will use the custom rH from SET_RH HC05 command when the command is received
- Added a help command to the HC05 commands to provide insight on what each command does

My discord is p.penguin if you have any comments, questions, concerns, etc. :D

ifBars committed:
Create README.md

ifBars committed:
Update README.md

ifBars committed:
Improving general readability and usability

- The major change is the comments added, some to specific lines, some to groups
- Some hard coded values like, the fan's minimum and maximum speeds, along with a few others, are now stored as float variables
- Some variable names were changed to improve readability, like h for example was changed to currentHumidity, as that is what it represents
- The SET_TEMP command has been split into 2 separate commands, SET_TEMPF and SET_TEMPC, for celsius and fahrenheit, to allow for more usability worldwide
- Conversion of celsius to fahrenheit is handled in a separate function to allow for modular usage in the SET_TEMPF and SET_TEMPC commands

ifBars committed:
Update Arduino IDE Code V1

Fixed a few typos where "minFanSpeed" was instead "minFanSped"

ifBars committed:
Update README.md

ifBars committed:
Delete README.md

ifBars committed:
Update Arduino IDE Code V1

ifBars committed:
Create new README.md

- Combined the original README from the GitHub repo, and the documentation provided in the discord, to create a new and improved README

ifBars commented:
Not sure why but it says the README is conflicting. Maybe because I forked the repo before the original README was added. Anyways, I think if you copy and paste the new README into the current one, it will allow the branches to be merged. If you do not like the new README for any reason let me know and I can update/remove it and you can just merge the code.

sudofizzz commented:
> Not sure why but it says the README is conflicting. Maybe because I forked the repo before the original README was added. Anyways, I think if you copy and paste the new README into the current one, it will allow the branches to be merged. If you do not like the new README for any reason let me know and I can update/remove it and you can just merge the code.

Go ahead please!

ifBars committed:
Delete README.md

ifBars committed:
Fixing a few issues

- Removed README as talked with in dms with Sudofizz
- Updated IDE Code as I messed up SET_TEMP commands and made both SET_TEMPF and another slight overlook, however they are now fixed accordingly

ifBars committed:
Average temp and humid per hour

- Added a feature to track the average temp and humidity in the past hour, which is displayed upon the DATAREQ command.

ifBars committed:
Delete README

As talked in DMs with Sudofizz, manually updating instead

ifBars committed:
General fixes

- Added more comments and improved some others
- fixed minFanSpeed variable, again

ifBars committed:
Improvement to state changes

- Introduced a small, adjustable, 3 second delay between state changes (Ex: turning on/off devices or adjusting fan speed). This prevents rapid toggling and allows devices to stabilize before making further adjustments.
- Changed the hysteresis values to constants to allow for reusability and easy adjustability

ifBars committed:
Update Arduino IDE Code V1

- Organized Hysteresis variables
- Unsigned samplesPerHour to prevent negative numbers
- Added debug messages that are togglable with the debug boolean on line 31
- Added an adjustable delay in state changes
- Added a function to format DateTime to a string
- Added a function to calculate current dew point, and also returns dew point in DATAREQ command
- Updated code to make sure it will use the custom rH from SET_RH HC05 command
- Added a help command to the HC05 commands to provide insight on what each command does
